attestation-agent/attestation-agent/src/eventlog/mod.rs

```diff
@@ -62,124 +93,141 @@ impl EventLog {
-            // and do then unfinished RTMR extending.
-            match aael.integrity_check(&rtmr) {
-                true => debug!("Existing RTMR is consistent with current AAEL"),
-                false => {
-                    debug!(
-                        "Existing RTMR is not consistent with current AAEL, do a RTMR extending..."
-                    );
-                    let digest = match aael.events.is_empty() {
-                        true => alg.digest(
-                            format!(
-                                "INIT {}/{:0>width$}",
-                                aael.hash_algorithm,
-                                hex::encode(aael.init_state),
-                                width = aael.hash_algorithm.digest_len()
-                            )
-                            .as_bytes(),
-                        ),
-                        false => alg.digest(aael.events[0].as_bytes()),
-                    };
-                    rtmr_extender
-                        .extend_runtime_measurement(digest, pcr)
-                        .await
-                        .context("Extend RTMR failed")?;
+                })
                 }
-            }
-
-            let file = OpenOptions::new()
-                .append(true)
-                .open(EVENTLOG_PATH)
-                .context("open eventlog")?;
-
-            return Ok(Self {
-                writer: Box::new(FileWriter { file }),
-                rtmr_extender,
-                alg,
-                pcr,
-            });
-        }
 
-        debug!("No AA eventlog exists, creating a new one and do INIT entry recording...");
-        let file = File::create(EVENTLOG_PATH).context("create eventlog")?;
-        let writer = Box::new(FileWriter { file });
-        let mut eventlog = Self {
+            Err(_) => bail!("Failed to read wal cache. This is a significant error caused by a previous crash. Please try delete `{WAL_CACHE}` and restart the attestation agent."),
+            Ok(None) => Ok(Self {
             writer,
             rtmr_extender,
             alg,
             pcr,
-        };
-        eventlog
-            .extend_init_entry()
-            .await
-            .context("extend INIT entry")?;
-        Ok(eventlog)
+        })
+        }
     }
 
-    pub async fn extend_entry(&mut self, log_entry: LogEntry<'_>, pcr: u64) -> Result<()> {
-        let digest = log_entry.digest_with(self.alg);
-        // The order must be ensured to keep consistency. s.t. first write AAEL
-        // and then extend RTMR.
-        self.writer.append(&log_entry).context("write log entry")?;
-        self.rtmr_extender
-            .extend_runtime_measurement(digest, pcr)
-            .await?;
+    /// Record the event and the target digest into cache file before write, this would do
+    /// help when there is a crush between extending PCR and logging event.
+    fn wal_cache(&self, wal_cache: WalCache) -> Result<()> {
```

Comment:
what about:
```suggestion
    fn write_wal_cache(&self, wal_cache: WalCache) -> Result<()> {
```

since we have `read_wal_cache` too.

Reply:
nice suggestion. Let me fix